### PR TITLE
feat(container): move main section router to main container

### LINKED ISSUE #383: fix(main-ui): going to smart contracts section shows an error (the first time)
**Describe the bug**
When moving to the "Smart Contracts" section for the first time, this error appears in the console:

`
Warning: React does not recognize the `computedMatch` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `computedmatch` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in div (created by SmartContracts)
    in Switch (created by SmartContracts)
    in SmartContracts (created by Route)
    in Route (created by PropsRoute)
    in PropsRoute (created by MainPage)
    in Switch (created by MainPage)
    in div (created by MainPage)
    in div (created by MainPage)
    in MainPage (created by Route)
    in Route (created by PropsRoute)
    in PropsRoute (created by Component)
    in Switch (created by Component)
    in div (created by App)
    in App (created by Component)
    in Component (created by Root)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by Connect(ConnectedRouter))
    in Connect(ConnectedRouter) (created by Root)
    in Provider (created by Root)
    in Root (created by HotExportedRoot)
    in AppContainer (created by HotExportedRoot)
    in HotExportedRoot
`

**To Reproduce**
1. Go to Main page in Sheikah
2. Click on Smart Contracts section

**Expected behavior**
No error in console

### PR DESCRIPTION
**Pull request by mmartinbar**
# PR Prelude

Thank you for contributing to sheikah! :)

**Please fill in this template (by putting an `x` inside
the brackets) _before_ filing your PR:**

- [x] I have read and understood [CODE_OF_CONDUCT][code] document.
- [x] I have read and understood [CONTRIBUTING][cont] document.
- [x] I have read and understood [STYLEGUIDE][style] document.
- [x] I have included tests for the changes in my PR. If not, I have included a
  rationale for why I haven't.
- [x] My code is formatted and is accepted by `yarn fmt`.
- [x] **I understand my PR may be closed if it becomes obvious I didn't
  actually perform all of these steps.**

# Why this change is necessary and useful

Main container now holds the main routing logic (both the sections routing and the tabs routing). The main container and not the main page component is the one that is being rendered from the routes. Main container is now pre-connected with redux for future use.

Additionally, URLs used under /main have been moved to the URL constants file (re #329). Also, the refactor of the smart contracts section fixes #383.

[code]: https://github.com/witnet/sheikah/blob/master/.github/CODE_OF_CONDUCT.md
[cont]: https://github.com/witnet/sheikah/blob/master/.github/CONTRIBUTING.md
[style]: https://github.com/witnet/sheikah/blob/master/docs/STYLEGUIDE.md

### CODE REVIEW

File: app/renderer/ui/components/main/sections/attestations/index.tsx [hunk 1]
```diff
@@ -23,6 +25,7 @@ class Attestations extends React.Component<SectionProps> {
 const AttestationsSection: SectionInfo = {
   key: "attestations",
   caption: "Attestations",
+  link: urls.ATTESTATIONS_SECTION,
```
Comment (Tommytrg) [inline app/renderer/ui/components/main/sections/attestations/index.tsx]: What do you think about rename "link" prop as "pathname"?

File: app/renderer/ui/components/main/sections/blockExplorer/index.tsx [hunk 2]
```diff
@@ -23,6 +25,7 @@ export class BlockExplorer extends React.Component<SectionProps> {
 const BlockExplorerSection: SectionInfo = {
   key: "blockexplorer",
   caption: "Block Explorer",
+  link: urls.BLOCK_EXPLORER_SECTION,
```
Comment (Tommytrg) [inline app/renderer/ui/components/main/sections/blockExplorer/index.tsx]: What do you think about rename "link" prop as "pathname"?

File: app/renderer/ui/components/main/sections/community/index.tsx [hunk 3]
```diff
@@ -23,6 +25,7 @@ export class Community extends React.Component<SectionProps> {
 const CommunitySection: SectionInfo = {
   key: "community",
   caption: "Community",
+  link: urls.COMMUNITY_SECTION,
```
Comment (Tommytrg) [inline app/renderer/ui/components/main/sections/community/index.tsx]: What do you think about rename "link" prop as "pathname"?

File: app/renderer/ui/components/main/sections/index.tsx [hunk 4]
```diff
@@ -52,6 +52,7 @@ export class SectionComponent<Props> extends React.Component<SectionProps & Path
 export interface SectionInfo {
   key: string
   caption: string
+  link: string
```
Comment (Tommytrg) [inline app/renderer/ui/components/main/sections/index.tsx]: What do you think about rename "link" prop as "pathname"?

File: app/renderer/ui/components/main/sections/index.tsx [hunk 5]
```diff
@@ -86,5 +87,6 @@ export class TabComponent<Props> extends React.Component<TabProps & PathNameProp
 export interface TabInfo {
   key: string
   caption: string
+  link: string
```
Comment (Tommytrg) [inline app/renderer/ui/components/main/sections/index.tsx]: What do you think about rename "link" prop as "pathname"?

File: app/renderer/ui/components/main/sections/smartContracts/index.tsx [hunk 6]
```diff
@@ -56,6 +48,7 @@ class SmartContracts extends React.Component<SmartContractsProps> {
 const SmartContractsSection: SectionInfo = {
   key: "smartcontracts",
   caption: "Smart Contracts",
+  link: urls.SMART_CONTRACTS_SECTION,
```
Comment (Tommytrg) [inline app/renderer/ui/components/main/sections/smartContracts/index.tsx]: What do you think about rename "link" prop as "pathname"?

File: app/renderer/ui/components/main/sections/smartContracts/tabs/easyComposer/index.tsx [hunk 7]
```diff
@@ -22,6 +23,7 @@ class EasyComposer extends TabComponent<any> {
 const EasyComposerTab: TabInfo = {
   key: "easycomposer",
   caption: "easy composer",
+  link: urls.EASY_COMPOSER_TAB,
```
Comment (Tommytrg) [inline app/renderer/ui/components/main/sections/smartContracts/tabs/easyComposer/index.tsx]: What do you think about rename "link" prop as "pathname"?

File: app/renderer/ui/components/main/sections/smartContracts/tabs/marketplace/index.tsx [hunk 8]
```diff
@@ -50,6 +52,7 @@ class MarketPlace extends TabComponent<any> {
 const MarketPlaceTab: TabInfo = {
   key: "MarketPlace",
   caption: "Marketplace",
+  link: urls.MARKETPLACE_TAB,
```
Comment (Tommytrg) [inline app/renderer/ui/components/main/sections/smartContracts/tabs/marketplace/index.tsx]: What do you think about rename "link" prop as "pathname"?

File: app/renderer/ui/components/main/sections/smartContracts/tabs/myContracts/index.tsx [hunk 9]
```diff
@@ -19,10 +20,11 @@ class MyContracts extends TabComponent<any> {
   }
 }
 
-const TransactionsTab: TabInfo = {
+const MyContractsTab: TabInfo = {
   key: "mycontracts",
   caption: "My contracts",
+  link: urls.MY_CONTRACTS_TAB,
```
Comment (Tommytrg) [inline app/renderer/ui/components/main/sections/smartContracts/tabs/myContracts/index.tsx]: What do you think about rename "link" prop as "pathname"?

File: app/renderer/ui/components/main/sections/smartContracts/tabs/proEditor/index.tsx [hunk 10]
```diff
@@ -19,10 +20,11 @@ class TabProEditor extends TabComponent<any> {
   }
 }
 
-const TransactionsTab: TabInfo = {
+const ProEditorTab: TabInfo = {
   key: "proEditor",
   caption: "PRO Editor",
+  link: urls.PROEDITOR_TAB,
```
Comment (Tommytrg) [inline app/renderer/ui/components/main/sections/smartContracts/tabs/proEditor/index.tsx]: What do you think about rename "link" prop as "pathname"?

File: app/renderer/ui/components/main/sections/wallet/index.tsx [hunk 11]
```diff
@@ -26,41 +26,15 @@ class Wallet extends React.Component<WalletProps> {
       {
         caption: tab.caption,
         key: tab.key,
-        link: `/main/wallet/${tab.key}`
+        link: `${tab.link}`
```
Comment (Tommytrg) [inline app/renderer/ui/components/main/sections/wallet/index.tsx]: What do you think about rename "link" prop as "pathname"?

File: app/renderer/ui/containers/WalletForm.tsx [hunk 12]
```diff
@@ -560,40 +560,38 @@ class WalletFormContainer extends
...
-              render={this.renderImportXprv}
-            />
-            <Route
-              path="/"
-              render={this.renderWelcome}
-            />
-          </Switch>
-        </WalletForm>
-      </>
+      <WalletForm spinner={this.state.spinner}>
```
Comment (Tommytrg) [inline app/renderer/ui/containers/WalletForm.tsx]: :ok_hand:

File: app/renderer/ui/components/main/sections/attestations/index.tsx (hunk 1 above)
Comment (aesedepece) [reply app/renderer/ui/components/main/sections/attestations/index.tsx]: I'd go for `path`.

Comment (mmartinbar) [reply app/renderer/ui/components/main/sections/attestations/index.tsx]: Path it is :+1:

**Pull request merged**